packages/astro/src/astro-components/unstyled/SignInButton.astro

```diff
@@ -0,0 +1,49 @@
+---
+import type { HTMLTag, Polymorphic } from 'astro/types'
+import type { SignInProps } from "@clerk/types";
+type Props<Tag extends HTMLTag> = Polymorphic<SignInProps & { as: Tag; id?: string; mode?: 'redirect' | 'modal' }>
+
+import { customAlphabet, urlAlphabet } from "nanoid";
+
+const safeId = customAlphabet(urlAlphabet, 10)();
+
+const {
+  as: Tag = 'button',
+  id = safeId,
+  forceRedirectUrl,
+  fallbackRedirectUrl,
+  signUpFallbackRedirectUrl,
+  signUpForceRedirectUrl,
+  mode,
+  ...elementProps
+} = Astro.props
+
+const signInOptions = {
+  forceRedirectUrl,
+  fallbackRedirectUrl,
+  signUpFallbackRedirectUrl,
+  signUpForceRedirectUrl,
+};
+---
+
+<Tag {...elementProps} id={id}>
+  <slot>Sign in</slot>
+</Tag >
+
+<script is:inline define:vars={{ signInOptions, mode, safeId }}>
+  const btn = document.querySelector(`#${safeId}`);
```

Comment:
Question about the usage of `id` here - is it safe to use the `id` as selector or should we use a `data-*` attribute? `id` has a big chance of getting replaced here since this is an unstyled component.

Reply:
@panteliselef If we go this way we will need to update the [mounting function](https://github.com/clerk/javascript/blob/5c3adbbb3e358182ae307beec5f67f35a79ca51c/packages/astro/src/client/mount-clerk-astro-js-components.ts#L18) since it looks for all `data-clerk-id` elements.

What about

- `data-clerk-interactive-ui` for the interactive components
- `data-clerk-unstyled-ui` for the unstyled

or we can drop the `ui` in there as well

Reply:
No actually, `data-clerk-ui` makes sense since it's a UI component. `data-clerk-unstyled` sounds more better

Reply:
Updated 🚀